From 929bc291717c611b629fd1969b6320431a7bfe58 Mon Sep 17 00:00:00 2001
From: David Vrabel <david.vrabel@citrix.com>
Date: Tue, 27 Oct 2015 11:29:15 +0000
Subject: [PATCH] x86/xen: add reschedule point when mapping foreign GFNs

Mapping a large range of foreign GFNs can take a long time, add a
reschedule point after each batch of 16 GFNs.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
diff --git a/arch/x86/xen/mmu.c b/arch/x86/xen/mmu.c
index 09f58e37..897653d 100644
--- a/arch/x86/xen/mmu.c
+++ b/arch/x86/xen/mmu.c
@@ -2490,6 +2490,8 @@ static int __xen_remap_domain_mfn_range(struct mm_struct *mm,
 			index += done;
 		} while (batch_left);
 
+		cond_resched();
+
 		nr -= batch;
 		addr += range;
 	}
